Issue21

bjorgan opened the pull request:
Test data generation and tests for sun, moon and aos/los.

bjorgan committed:
Preliminary, ugly bash script for generating test data using predict's UDP server interface.

bjorgan committed:
Modified predict's UDP client sample code.

bjorgan committed:
Some comments in generation script

bjorgan committed:
Added QTH and TLE file for testcase generation.

bjorgan committed:
Changed directory for generation script. Changed data format to comma-separated.

bjorgan committed:
Fixed N/W to N/E conversion in testcase generation script.

bjorgan committed:
Fixes in testcase executable after ryeng's API changes.

bjorgan committed:
Autogeneration of output filename

bjorgan committed:
Changes in thresholds for tests.

bjorgan committed:
Coordinate fix in sun/moon testcase generation.

bjorgan committed:
Threw around compare functions, moved validity tests from loadFile to different functions (in order to facilitate for the absence of e.g. TLE)

bjorgan committed:
Added test executables for moon and sun files (due to the nature of the function calls: Code reusage difficult, copy and replace was necessary).

bjorgan committed:
Changed lower bound for values from time - DIFF to time (since the time error would only be within one second added to the unix epoch)

bjorgan committed:
Moved Main.cpp in test to TestSatellite.cpp.

bjorgan committed:
Moved tests to test directory (and renamed the files).

bjorgan committed:
Added some initial testcase files in lieu of more carefully selected cases.

bjorgan committed:
Added some notes at the top of the testcase generation script.

bjorgan committed:
Moved test data generation script (and related) to different directory from the test data itself.

bjorgan committed:
Added orbit, sun and moon tests to CTest.

bjorgan committed:
Added a sanity check for AOS/LOS: Loops through the orbit until AOS/LOS to check whether the times make sense.

bjorgan committed:
C++ flags in CMakeLists.txt.

bjorgan committed:
Changed from C++11 to C++0x

bjorgan committed:
Style changes in order to conform with thomas' style choices.

ryeng commented:
It generally looks good. Great work putting this test system together!

Since I haven't looked at the code before, I try to understand what things are based on file, function and class names. The `testcase.[h|cpp]` files contain a test case reader that parses the file containing test parameters and results. I suggest we rename the files (and maybe the class if  you feel it makes sense) to, e.g., `test_reader.[h|cpp]`, `testcase_reader.[h|cpp]`, or something similar.

A few suggestions about the paths and file names:
- `mv tests/test_data tests/data`
- `mv tests/test_data_generation tests/data-generator` (or something similar)
- The prefix `testcase_` in test case file names is redundant. How about `satellite_*.test`, `sun_*.test` and `moon_*.test`?

I hope these renames would help reduce the risk of carpal tunnel syndrome in the future. :-)

bjorgan committed:
Renamed testcase.cpp to testcase_reader.cpp.

bjorgan committed:
Shortened test names.

bjorgan committed:
Removed orbit- prefix from orbit tests.

ryeng committed:
Temporarily disable Doxygen generation

Travis-CI has moved to a new container-based infrastructure, but there's
no option for installing packages like Doxygen yet. The old
infrastructure still exists, but with less resources, so builds take
forever.

See http://docs.travis-ci.com/user/migrating-from-legacy/ for details.

bjorgan committed:
Changed testfile prefix and postfix.

bjorgan committed:
Renamed test data folder and test data generation folder.

bjorgan committed:
Renamed TestCase to TestCaseReader.

bjorgan commented:
I've renamed testcase so that it is more clear that it is a reader, and shortened test names, testcase filenames and changed the folder names. Carpal tunnel syndrome: Avoided
